File: 2022/4.py
import sys

f = open(sys.argv[1], "r")
lines = [line.strip() for line in f.readlines()]
answer = 0

for idx, line in enumerate(lines):
    a, b, c, d = [int(y) for x in line.split(",") for y in x.split("-")]

    if (c <= b and d >= a) or (a <= d and b >= c):
        answer = answer + 1

print(answer)

File: 2022/4.js
const { readFileSync } = require('fs')
const { argv } = require('process')

let lines = readFileSync(argv[2]).toString().split(/\r?\n/).slice(0, -1)
let answer = 0

for (const [idx, line] of lines.entries()) {
    let [a, b, c, d] = line.split(",").map((x) => x.split("-")).flat().map(x => parseInt(x))

    if ((b >= c && a <= d) || (d >= a && c <= b))
        answer++
}

console.log(answer)
